graphblas/select/__init__.py

```diff
@@ -0,0 +1,61 @@
+# All items are dynamically added by classes in operator.py
+# This module acts as a container of SelectOp instances (which are
+# identical to IndexUnaryOp under the hood, except they must return
+# boolean and they default to `select` when called, rather than
+# `apply`).
+_delayed = {}
+from graphblas import operator  # noqa isort:skip
+
+del operator
+
+
+def __dir__():
+    return globals().keys() | _delayed.keys()
+
+
+def __getattr__(key):
+    if key in _delayed:
+        func, kwargs = _delayed.pop(key)
+        rv = func(**kwargs)
+        globals()[key] = rv
+        return rv
+    raise AttributeError(f"module {__name__!r} has no attribute {key!r}")
+
+
+def _resolve_expr(expr, callname, opname):
+    from graphblas.base import BaseExpression
+
+    if not isinstance(expr, BaseExpression) and hasattr(expr, "select"):
```

Comment:
What's with `and hasattr(expr, "select")` here?  Did you mean `or not hasattr(expr, "select")`?  It may be clearer to check explicitly for `MatrixExpression` and `VectorExpression`.

Reply:
I was trying to catch `ScalarExpression`, but you're right that explicitly calling out `MatrixExpression` and `VectorExpression` will be more readable.